doc/CMakeLists.txt

```
@@ -10,47 +51,22 @@ endif()
 
 find_package(Doxygen 1.8.0 ${REQUIRED})
 
-if(DOXYGEN_FOUND)
-  cmake_path(RELATIVE_PATH CMAKE_SOURCE_DIR BASE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} OUTPUT_VARIABLE MIR_SOURCE_RELATIVE_PATH)
-  if(NOT ${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
-    file(
-        COPY ${CMAKE_CURRENT_SOURCE_DIR}/sphinx/
-        DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/sphinx/
-        PATTERN *.in EXCLUDE
-    )
-  endif()
 
-  configure_file(
-      ${CMAKE_CURRENT_SOURCE_DIR}/sphinx/Doxyfile.in
-      ${CMAKE_CURRENT_BINARY_DIR}/sphinx/Doxyfile @ONLY IMMEDIATE
-  )
+if(DOXYGEN_FOUND)
+  set(WORKING_DIR ${CMAKE_CURRENT_BINARY_DIR}/sphinx)
 
   configure_file(
-      ${CMAKE_CURRENT_SOURCE_DIR}/sphinx/reference/introducing_the_miral_api.md.in
-      ${CMAKE_CURRENT_BINARY_DIR}/sphinx/reference/introducing_the_miral_api.md @ONLY IMMEDIATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/sphinx/Doxyfile.in
+    ${WORKING_DIR}/Doxyfile @ONLY IMMEDIATE
   )
 
-  # use the starter pack Makefile
-  add_custom_command(
-      OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/sphinx/.sphinx/venv
-      COMMAND make -C ${CMAKE_CURRENT_BINARY_DIR}/sphinx install
+  add_custom_target(doxygen
+    COMMAND doxygen ${WORKING_DIR}/Doxyfile
+    BYPRODUCTS ${WORKING_DIR}/doxygen_output/xml 
   )
 
-  if(BUILD_DOXYGEN)
-    set(ALL "ALL")
-  endif()
-
-  add_custom_target(doc ${ALL}
-      COMMAND make -C ${CMAKE_CURRENT_BINARY_DIR}/sphinx html
-      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/sphinx/.sphinx/venv
-              guides
-      BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/sphinx/_build
-                 ${CMAKE_CURRENT_BINARY_DIR}/sphinx/xml
-  )
-
-  install(
-      DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/sphinx/_build/
-      DESTINATION ${CMAKE_INSTALL_PREFIX}/share/doc/mir-doc
-      OPTIONAL
-  )
+  add_custom_target(doxygen-clean
+    COMMAND rm -r ${WORKING_DIR}/doxygen_output/xml ${CMAKE_SOURCE_DIR}/doc/sphinx/api
```

I think we want to kill `doxygen_output` itself, too?
```suggestion
    COMMAND rm -r ${WORKING_DIR}/doxygen_output ${CMAKE_SOURCE_DIR}/doc/sphinx/api
```